libs/sdk-core/src/breez_services.rs

```diff
@@ -574,6 +574,27 @@ impl BreezServices {
         Ok(self.persister.get_payment_by_hash(&hash)?)
     }
 
+    /// Fetch the external metadata of a payment.
+    /// Useful for persisting external, platform-specific information,
+    /// unrelated to the SDK
+    pub async fn payment_external_metadata<T: serde::de::DeserializeOwned>(
```

Comment:
I think we need to add a "metadata" field to the payment structure (returned by list_payments). Otherwise the user would have to iterate over the list and fetch one by one.
If we do so the advantage is that we can remove this method as we already have `payment_by_hash`.